test/internal/transport/websocket/WebSocketMessageClientTest.ts

```diff
@@ -42,4 +36,161 @@ describe("WebSocketMessageClient", () => {
         assert(clean([""]).length === 0);
     });
 
+    it("correctly format user message", done => {
+        const corrId = guid();
+        const client = new WebSocketEventMessageClient(
+            {
+                data: {},
+                extensions: { team_id: "Txxxxxxx", correlation_id: corrId, operationName: "Foor" },
+                secrets: [],
+            }, { send: () => { //
+                // Intentionally left empty
+            } } as any as WebSocket );
+
+        const msg: SlackMessage = {
+            attachments: [{
+                fallback: "test",
+                text: "test",
+                actions: [
+                    buttonForCommand({text: "Foo"}, "HelloWorld", {name: "cd"}),
+                ],
+            }],
+        };
+
+        client.send(msg, { userAgent: "slack", team: "Txxxxxxx", users: ["cd", "rod"] } as SlackDestination, { id: "123456"})
+            .then(fm => {
+                assert(fm.api_version === "1");
+                assert(fm.correlation_id === corrId);
+                assert(fm.content_type === "application/x-atomist-slack+json");
+                assert(fm.team.id === "Txxxxxxx");
+                assert(fm.event === "Foor");
+                assert(fm.id === "123456");
+                assert(fm.destinations.length === 2);
+                assert(fm.destinations[0].user_agent === "slack");
+                assert(fm.destinations[0].slack.team.id === "Txxxxxxx");
+                assert(fm.destinations[0].slack.user.name === "cd");
+                assert(fm.destinations[1].user_agent === "slack");
+                assert(fm.destinations[1].slack.team.id === "Txxxxxxx");
+                assert(fm.destinations[1].slack.user.name === "rod");
+                assert(fm.actions.length === 1);
+                assert(fm.actions[0].id === "helloworld-0");
+                assert(fm.actions[0].command === "HelloWorld");
+                assert(fm.actions[0].parameters.length === 1);
+                assert(fm.actions[0].parameters[0].name === "name");
+                assert(fm.actions[0].parameters[0].value === "cd");
+                done();
+            });
+
+    }).timeout(5000);
```

Comment:
Are these tests actually hitting the Slack API and thus require the longer than default timeout?